editor-components/feature/feature13/feature13.tsx

```diff
@@ -1,209 +1,407 @@
+              key: "percentage",
+              displayer: "Percentage",
+              value: 85
+            }
+          ]
         },
-      ],
-    });
+      ]
+    })
+
+    let x = setInterval(() => {
+      this.sliderNext()
+    }, 3000);
   }
 
   getName(): string {
     return "Feature 13";
   }
 
+  activeTab: number = 0;
+
+  switchTab(tabIndex: number) {
+    this.activeTab = tabIndex;
+  }
+
+  sliderNext = () => {
+    const updatedImages = this.getComponentState("slider-images").map((image: SliderImage) => {
+      if (image.imageIndex === this.getComponentState("slider-images").length - 1)
+        return {
+          ...image,
+          imageIndex: 0
+        }
+      else
+        return {
+          ...image,
+          imageIndex: image.imageIndex + 1
+        }
+    })
+    this.setComponentState("slider-images", updatedImages)
+  }
+
+  sliderPrev = () => {
+    const updatedImages = this.getComponentState("slider-images").map((image: SliderImage) => {
+      if (image.imageIndex === 0)
+        return {
+          ...image,
+          imageIndex: this.getComponentState("slider-images").length - 1
+        }
+      else
+        return {
+          ...image,
+          imageIndex: image.imageIndex - 1
+        }
+    })
+    this.setComponentState("slider-images", updatedImages)
+  }
+
   render() {
+
+    if (JSON.stringify(this.getComponentState("slider-images-temp")) !== JSON.stringify(this.castToObject<SliderImage[]>("slider-images"))) {
+      this.setComponentState("slider-images-temp", this.castToObject<SliderImage[]>("slider-images"));
+      this.setComponentState("slider-images", this.castToObject<SliderImage[]>("slider-images"));
+    }
+
+
     return (
-      <div className={this.decorateCSS("container")}>
+      <div className={this.decorateCSS("container")} >
         <div className={this.decorateCSS("max-content")}>
-          <div className={this.decorateCSS("content")}>
-            <div className={this.decorateCSS("message")}>
-              <h1 className={this.decorateCSS("text")}>
-                {this.getPropValue("text")[0].value}
-              </h1>
-              <p className={this.decorateCSS("text-p")}>
-                {this.getPropValue("text")[1].value}
-              </p>
-
-              <ComposerLink path={this.getPropValue("text")[3].value}>
-                <span className={this.decorateCSS("button")}>
-                  {this.getPropValue("text")[2].value}
-                </span>
-              </ComposerLink>
-            </div>
-            <div className={this.decorateCSS("cards")}>
-              {this.castToObject<CardMultiple[]>("multiple-card").map(
-                (multiple: any, index: number) => (
-                  <div className={this.decorateCSS("multiple")} key={index}>
+          <div className={this.decorateCSS("comp-container")}>
+            <header className={this.decorateCSS("comp-header")}>
+              {this.getPropValue("title") &&
```

Comment:
you should use castToString to checking string values.  cause this.getPropValue returns a HTML tag